### ISSUE TITLE
Looks like recent IndexedDBShim doesn't work with db.js

### ISSUE THREAD

**Issue by kir** [indexeddbshim/IndexedDBShim#206]
Due to this, could not upgrade from IndexedDBShim 0.x to 2.0 in my project. 
Looks like IndexedDBShim doesn't like passing 'null' instead of undefined as keyRange parameters and such.

Even after that, put(object) without explicit key to a store with keyPath: 'id' didn't work either for me, but it is probably a separate bug.

**Comment by JamesMessinger** [comment]
The `0.x` version of IndexedDBShim was not consistent with native IndexedDB implementations in browsers, which made for a pretty bad polyfill.  :)   So we've made a **ton** of changes in `1.x` and `2.x` to make IndexedDBShim behave the same as native IndexedDB implementations.  However, there are a few places where different browsers have different behaviors.  In those cases, we implemented the behavior that was used by the most browsers.   

So, it's possible that the issues you're experiencing are due to different behaviors between browsers.  This is especially the case when it comes to values like `undefined`, `null`, `NaN`, empty arrays, empty strings, etc.   For example, take a look at [this unit test](https://github.com/axemclion/IndexedDBShim/blob/28cf794b8a3c097f94497c1ea341dca4cfe81acb/tests/IDBIndex/openCursor-spec.js#L402-L420).
